File: clients/python/examples/screen.py
"""
屏幕操作示例
"""

from wingman import WingmanClient
from PIL import Image
import io

with WingmanClient() as client:
    # 获取屏幕尺寸
    size = client.screen.get_size()
    w, h = size['width'], size['height']
    print(f"屏幕: {w}x{h}")

    # 截取全屏
    print("正在截图...")
    png_data = client.screen.capture()
    image = Image.open(io.BytesIO(png_data))
    image.save('screenshot.png')
    print("截图已保存到 screenshot.png")

    # 截取指定区域
    region_data = client.screen.capture(0, 0, 500, 500)
    region_image = Image.open(io.BytesIO(region_data))
    region_image.save('screenshot_region.png')
    print("区域截图已保存到 screenshot_region.png")

    # 获取像素颜色
    pixel = client.screen.get_pixel(100, 100)
    print(f"像素 (100, 100) 颜色: RGB({pixel['r']}, {pixel['g']}, {pixel['b']})")

    # 查找颜色 (红色: 0xFF0000)
    found = client.screen.find_color(0xFF0000, 0, 0, w, h, tolerance=10)
    if found:
        print(f"找到红色位置: ({found['x']}, {found['y']})")
    else:
        print("未找到红色")

    # 查找多个颜色点
    points = client.screen.find_colors(0xFF0000, 0, 0, w, h, tolerance=10, count=10)
    print(f"找到 {len(points)} 个红色点")
    for i, p in enumerate(points[:5]):
        print(f"  [{i+1}] ({p['x']}, {p['y']})")

    # 查找图像
    try:
        found = client.screen.find_image('target.png', 0, 0, w, h, threshold=0.9)
        if found:
            print(f"找到图像位置: ({found['x']}, {found['y']})")
        else:
            print("未找到图像")
    except FileNotFoundError:
        print("请准备 target.png 文件进行图像匹配测试")

File: clients/node/examples/screen.ts
/**
 * 屏幕操作示例
 */

import { WingmanClient } from '@wingman/client';
import * as fs from 'fs';

async function main() {
  const client = new WingmanClient();

  client.on('connected', () => console.log('已连接'));
  client.on('disconnected', () => console.log('已断开'));

  await client.connect();

  // 获取屏幕尺寸
  const size = await client.screen.getSize();
  console.log(`屏幕: ${size.width}x${size.height}`);

  // 截取全屏
  console.log('正在截图...');
  const screenshot = await client.screen.capture();
  fs.writeFileSync('screenshot.png', screenshot);
  console.log('已保存到 screenshot.png');

  // 截取区域
  const region = await client.screen.capture(0, 0, 500, 500);
  fs.writeFileSync('screenshot_region.png', region);
  console.log('已保存区域截图');

  // 获取像素颜色
  const pixel = await client.screen.getPixel(100, 100);
  console.log(`像素(100,100): RGB(${pixel.r}, ${pixel.g}, ${pixel.b})`);

  // 查找颜色
  const found = await client.screen.findColor(
    0xFF0000,  // 红色
    0, 0, size.width, size.height,
    10  // 容差
  );
  if (found) {
    console.log(`找到红色在: (${found.x}, ${found.y})`);
  } else {
    console.log('未找到红色');
  }

  // 查找多个颜色点
  const points = await client.screen.findColors(
    0xFF0000,
    0, 0, size.width, size.height,
    10, 100
  );
  console.log(`找到 ${points.length} 个红色点`);

  // 查找图像
  try {
    const imagePos = await client.screen.findImage(
      'target.png',
      0, 0, size.width, size.height,
      0.9
    );
    if (imagePos) {
      console.log(`找到图像在: (${imagePos.x}, ${imagePos.y})`);
    }
  } catch (e) {
    console.log('请准备 target.png 进行图像匹配测试');
  }

  await client.disconnect();
}

main().catch(console.error);
